cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientFactory.java

```diff
@@ -80,31 +85,26 @@ public CloudControllerRestClient createClient(URL controllerUrl, CloudCredential
 
     public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, String organizationName,
         String spaceName) {
-        this.oAuthClient = createOAuthClient(controllerUrl);
-        LoggregatorClient loggregatorClient = new LoggregatorClient(trustSelfSignedCerts);
-
-        return new CloudControllerRestClientImpl(controllerUrl, credentials, restTemplate, oAuthClient, loggregatorClient, organizationName,
-            spaceName);
+        return createClient(controllerUrl, credentials, organizationName, spaceName, createOAuthClient(controllerUrl));
     }
 
     public CloudControllerRestClient createClient(URL controllerUrl, CloudCredentials credentials, String organizationName,
         String spaceName, OAuthClient oAuthClient) {
-        this.oAuthClient = oAuthClient;
-        LoggregatorClient loggregatorClient = new LoggregatorClient(trustSelfSignedCerts);
+        CloudControllerRestClientImpl clientWithoutTarget = new CloudControllerRestClientImpl(controllerUrl, credentials, restTemplate,
+            oAuthClient, null, null);
+        CloudSpace target = clientWithoutTarget.getSpace(organizationName, spaceName);
```

Comment:
clientWithoutTarget can retrieve space o.O

Reply:
Yup, you can do `cf spaces` and `cf orgs` even if you don't have a target.